initial-promises/node-server/common/1.0/update.cf

```diff
@@ -186,155 +237,198 @@ bundle agent update_action
+        delete  => tidy,
+        comment => "Deleting ${g.rudder_ncf_hash_file} as common ncf files couldn't be downloaded";
+
+    (new_promises_available|rudder_ncf_hash_update_repaired).!root_server.rudder_ncf_local_update_error::
+      "${g.rudder_ncf}/local/${g.rudder_ncf_hash_file}"
+        delete  => tidy,
+        comment => "Deleting ${g.rudder_ncf_hash_file} as custom ncf files couldn't be downloaded";
+
+
+    root_server|(rudder_promises_generated_tmp_file_ok|(new_promises_available.(config|config_ok)).!no_update.!rudder_promises_generated_tmp_file_error)::
       # Every time we check update inputs successfully (already up to date or
       # updated), touch a file to let other promises know we are doing ok
       "${sys.workdir}/last_successful_inputs_update"
         touch      => "true";
 
     # Copy the tools only if the file rudder_tools_updated is not up to date
     !policy_server::
-      "${g.rudder_tools_updated}"
+      "${g.rudder_tools}/${rudder_tools_updated_tmp}"
         copy_from         => remote_unsecured_without_perms("${server_info.cfserved}", "${g.rudder_tools_updated_origin}"),
         action            => immediate,
         move_obstructions => "true",
         classes           => success("rudder_tools_updated_repaired", "rudder_tools_updated_error", "rudder_tools_updated_kept"),
         comment           => "Check if we need to update the tools";
 
-    # We copy only if we have the class rudder_tools_updated_repaired
-    !policy_server.rudder_tools_updated_repaired::
+    # We copy only if we have new tools based on the timestamp
+    !policy_server.new_tools_available::
       "${g.rudder_tools}"
         copy_from         => remote_unsecured("${server_info.cfserved}", "${g.rudder_tools_origin}"),
         depth_search      => recurse_ignore("inf", @{g.excludedreps}),
         file_select       => by_fullname(@{rudder_tools_files_to_select}),
         action            => immediate,
         move_obstructions => "true",
-        classes           => success("rudder_tools_updated", "rudder_tools_update_error", "rudder_tools_updated_ok");
+        classes           => success("rudder_tools_updated", "rudder_tools_update_error", "rudder_tools_updated_ok"),
+        comment           => "Update the Rudder tools";
 
     # If we couldn't update the tools, then we must force a copy at next run
     rudder_tools_update_error::
       "${g.rudder_tools_updated}"
         delete => tidy,
         comment => "Deleting ${g.rudder_tools_updated} as the tools couldn't be updated";
 
+  methods:
+    pass3.server_ok::
+      "any" usebundle => rudder_common_report("Common", "log_repaired", "hasPolicyServer-root@@common-root@@00", "Update", "None", "Started the server (cf-serverd)");
+    pass3.executor_ok::
+      "any" usebundle => rudder_common_report("Common", "log_repaired", "hasPolicyServer-root@@common-root@@00", "Update", "None", "Started the scheduler (cf-execd)");
```

Comment:
These logs should be in the update_reports bundle below. They will break reporting by being here (since this is run by failsafe.cf and our "start run" message is sent out at the beginning of promises.cf)